### PR TITLE
feat: override array as primitive when deep merge

### PR DESCRIPTION
**Pull request by popomore**
Ref https://github.com/eggjs/egg/issues/500

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -51,31 +43,25 @@ module.exports = function extend() {
...
-						} else {
-							clone = src && isPlainObject(src) ? src : {};
-						}
+			// Prevent never-ending loop
+			if (target === copy) continue;
 
-						// Never move original objects, clone them
-						target[name] = extend(deep, clone, copy);
+			// Recurse if we're merging plain objects
+			if (deep && copy && isPlainObject(copy)) {
```
Comment (dead-horse) [inline index.js]: tab 换成空格？

**Comment by popomore** [comment]
就第一个 commit 改了代码

**Comment by popomore** [comment]
tape 先不改了

**Review by dead-horse — APPROVED** [review]

**Comment by ljharb** [comment]
um, what? why is this forked?

**Comment by popomore** [comment]
@ljharb the array merging is not expected for us, it's like object, but it's strange when we merge two array config.

**Comment by fengmk2** [comment]
Or extend can accept an option to impl this feature? @ljharb

**Comment by popomore** [comment]
it's not easy to give another opt for node-extend

**Comment by popomore** [comment]
先合这个？

**Review by fengmk2 — APPROVED** [review]

**Comment by fengmk2** [comment]
@popomore 你来发

**Comment by popomore** [comment]
+ extend2@1.0.0

**Pull request merged**